logstash-core/src/main/java/org/logstash/plugins/AliasRegistry.java

```diff
@@ -37,13 +50,103 @@ PluginCoordinate withName(String name) {
+                LOGGER.warn("Malformed yaml file", badSyntaxExcp);
+                return Collections.emptyMap();
+            }
+
+            // calculate the hash-256
+            final String sha256Hex = DigestUtils.sha256Hex(yaml);
+            if (!sha256Hex.equals(extractedHash)) {
+                LOGGER.warn("Bad checksum value, expected {} but found {}", sha256Hex, extractedHash);
+                return Collections.emptyMap();
+            }
+
+            // decode yaml to nested maps
+            final Map<String, Map<String, String>> aliasedDescriptions;
+            try {
+                ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+                aliasedDescriptions = mapper.readValue(yaml, Map.class);
+            } catch (IOException ioex) {
+                LOGGER.error("Error decoding the yaml aliases file", ioex);
+                return Collections.emptyMap();
+            }
+
+            // convert aliases nested maps definitions to plugin alias definitions
+            final Map<PluginCoordinate, String> defaultDefinitions = new HashMap<>();
+            defaultDefinitions.putAll(extractDefinitions(PluginType.INPUT, aliasedDescriptions));
+            defaultDefinitions.putAll(extractDefinitions(PluginType.CODEC, aliasedDescriptions));
+            defaultDefinitions.putAll(extractDefinitions(PluginType.FILTER, aliasedDescriptions));
+            defaultDefinitions.putAll(extractDefinitions(PluginType.OUTPUT, aliasedDescriptions));
+            return defaultDefinitions;
+        }
+
+        private void parseYamlFile(String yamlResourcePath) {
+            final InputStream in = this.getClass().getClassLoader().getResourceAsStream(yamlResourcePath);
+
+            try (Scanner scanner = new Scanner(in, StandardCharsets.UTF_8.name())) {
+                // read the header line
+                final String header = scanner.nextLine();
+                if (!header.startsWith("#CHECKSUM:")) {
+                    throw new IllegalArgumentException("Bad header format, expected '#CHECKSUM: ...' but found " + header);
+                }
+                extractedHash = header.substring("#CHECKSUM:".length()).trim();
+
+                // read the comment
+                scanner.nextLine();
+
+                // collect all remaining lines
+                yaml = loadYamlSection(scanner).toString();
+            }
+        }
+
+        private StringBuilder loadYamlSection(Scanner scanner) {
+            final StringBuilder yaml = new StringBuilder();
+            while (true) {
+                final String line;
+                try {
+                    line = scanner.nextLine();
+                    yaml.append(line);
+                } catch (NoSuchElementException noMoreLinesEx) {
+                    break;
+                }
+                if (scanner.hasNext()) {
+                    yaml.append("\n");
+                }
+            }
+            return yaml;
```

Comment:
The boundary between `loadYamlSection ` and its caller is stateful on both ends -- it requres a `Scanner` in a certain state as input, and produces a stateful `StringBuilder` when a `String` will do. I am wary of methods that have this much state tie-in with their callers (even when `private`), because the hidden state can be the source of really hard-to-debug issues during later refactors.

That said, we can make the read-the-rest-from-the-scanner operation simpler, which I think obviates the need to break it out.

We can use `Scanner#useDelimiter("\\Z")` to cause the subsequent `Scanner#next()` to return everything up to -- but not including -- the trailing newline.

```suggestion

            final StringBuilder yamlBuilder = new StringBuilder();
            scanner.useDelimiter("\\Z") // EOF, less optional trailing newline
            if (scanner.hasNext()) {
              yamlBuilder.append(scanner.next());
            }
            yaml = yamlBuilder.toString();
```